### DISCUSSION TITLE
Questions about Restoring to official firmware.

### DISCUSSION THREAD

**Discussion by happyreadygo** [dangowrt/owrt-ubi-installer#194] [Q&A] [answered]
I have 11 of E8450.  I am practicing to restore from openwrt-23(UBI-installer-v1.0.2) to linksys firmware. I did following anything write on restoring section. At the end,It run linksys firmware  **but It can't remember any setting after reboot**. Everytime reboot into setup wizard of linksys firmware. Also got problem with nonUBI-openwrt_23  , However it's remember setting normally after I re-install openwrt(UBI-v1.0.2) .

I also have question about Backup stock bootchain .Why there're only 4 mtdx files(mtd0,mtd1,mtd2 and mtd3) in /tmp/boot_backup. Aren't stock and non-UBI openwrt have 15 mtdx files ( mtd0-14)? 

Last question,  How importand of preloader(bl2) and fip? , From what I understand(very little),u-boot is inside fip and it's required preloader. If something wrong about mtd0(bl2) and mtd1(fip) such as flash wrong file or flash file corrupted.  We can't access u-boot boot menu and the router is deadly bricked beyond uart-serial level. While we can easy going with mtd2(factory) and mtd3(UBI) .Am I wrong or missunderstand something?

**Comment by dangowrt** [comment, marked as answer]
> Everytime reboot into setup wizard of linksys firmware.

I never saw that happening. Try to flash the stock firmware using the stock firmware itself, that may resolve such issues.

> Why there're only 4 mtdx files(mtd0,mtd1,mtd2 and mtd3) in /tmp/boot_backup. Aren't stock and non-UBI openwrt have 15 mtdx files ( mtd0-14)?

The MTD partitions are (unlike partitions on a block device, like SSD, eMMC, microSD or USB pendrive) defined by the device tree of the firmware you are running rather than being stored on a specific block of the storage device itself. As such, the layout differs depending on which firmware you are booting. If you use v1.0.x installer with OpenWrt 23.05.x you have the whole flash chip covered by just 4 partitions (mtd0, mtd1, mtd2, mtd3). The vendor firmware defines many more partitions: some of them are statically defined at certain offsets of the flash, others are dynamically defined  at runtime. For example, the content of the `firmware` partition is also accessible via the split sub-partitions `kernel`, `rootfs` and `rootfs_data` at runtime.
tl;dr: the same areas of the flash chip are accessible multiple times when using the stock firmware, and are mapped into only 4 partitions when using OpenWrt 23.05.x.



> If something wrong about mtd0(bl2) and mtd1(fip) such as flash wrong file or flash file corrupted. We can't access u-boot boot menu and the router is deadly bricked beyond uart-serial level. While we can easy going with mtd2(factory) and mtd3(UBI) .Am I wrong or missunderstand something?

Yes, this is true for OpenWrt UBI bootchain. With the stock bootchain it's slightly different: Instead of ARM TrustedFirmware-A open source you got a proprietary Preloader which loads U-Boot with custom MediaTek image header. Then U-Boot loads ARM TrustedFirmware-A BL31 and launches the kernel. However, regarding your question: Yes, if anything goes wrong while flashing the bootloader you will have to use [`mtk_uartboot`](https://github.com/981213/mtk_uartboot) to recover (yes, that still works using serial, no need to soldier JTAG or anything like that).
